### DISCUSSION TITLE
Addressing mode support

### DISCUSSION THREAD

**Discussion by m-kru** [esynr3z/corsair#106] [Questions]
Does Corsair support different addressing modes? I do not mean byte and word addressing, but `compact`, `regalign` and `fullalign` from SystemRDL nomenclature. Do you always align address space size of particular slave to the power of 2?

**Comment by esynr3z** [comment]
There are `address_alignment` and `address_increment` parameters that allows to configure the subject.
https://corsair.readthedocs.io/en/latest/config.html#globcfg-section

[Currently](https://github.com/esynr3z/corsair/blob/14dc5d40a52f7efac07393d0a53623fc231f8f77/corsair/templates/regmap_verilog.j2#L262) implementation is quite straight-forward, so full address (all bits) is always compared. I've never tried but it looks like alignment in non-power-of-2 way is possible. However, it also depends on interface you use on the top of the map and how it deals with non-aligned accesses.

Could you please provide some minimal practical example what feature do you need or what behavior you're willing to achieve?
